testing_and_setup/compass/ocean/scripts/plots/plot_initial_state.py

```diff
@@ -65,45 +67,60 @@ def main():
     txt = txt + ' {:9.2e}'.format(np.min(var)) + \
         ' {:9.2e}'.format(np.max(var)) + ' ' + varName + '\n'
 
+    cellsOnEdge = ncfile.variables['cellsOnEdge']
+    cellMask = np.zeros((nCells, nVertLevels), bool)
+    edgeMask = np.zeros((nEdges, nVertLevels), bool)
+    for k in range(nVertLevels):
+        cellMask[:, k] = k < maxLevelCell
+        cell0 = cellsOnEdge[:, 0]-1
+        cell1 = cellsOnEdge[:, 1]-1
+        edgeMask[:, k] = np.logical_and(np.logical_and(cellMask[cell0, k],
+                                                       cellMask[cell1, k]),
+                                        np.logical_and(cell0 >= 0,
+                                                       cell1 >= 0))
+
     plt.subplot(3, 3, 4)
     varName = 'temperature'
-    var = ncfile.variables[varName]
-    plt.hist(np.ndarray.flatten(var[:]), bins=100, log=True)
```

Comment:
The mask takes care of of flattening the array in this case but a more efficient way of flattening a typical numpy array is to call its `ravel` method: `var[:].ravel()`